### DISCUSSION TITLE
How to set custom db variables?

### DISCUSSION THREAD

**Discussion by aarvay** [supabase/supabase#1746] [Questions] [answered]
Based on the discussion from [Twitter](https://twitter.com/aarvay/status/1397933498401316867?s=20) that me @kiwicopple had. 

I'll try to explain my use case here:

Supabase client will be called from several hostnames. And I want to set RLS policy based on the current hostname. Basically my hosts table contains list of valid hosts, if the current host is present in the hosts table, i want to set it as the current_hostname db variable so i can use it in policies. How can i verify and set the current_hostname? 

Hosts table definition:
```sql
CREATE TABLE hosts (
    id integer PRIMARY KEY,
    name text NOT NULL,
    ...
    ...
);
```

All other tables that need RLS will contain host_id as a foreign key. Now I want to set a policy like this:

```sql

CREATE POLICY on some_table for
    select using (current_host_id() = host_id);
```

EDIT: Actually there's nothing to verify, all I need is the access to current hostname in the policy. I can just do:

```sql

CREATE POLICY on some_table for
SELECT(
    USING host_id in (
        SELECT id from hosts where name = current_hostname()
    )
);
```

**Comment by steve-chavez** [comment, marked as answer]
Hey @aarvay,

> current_hostname()

How will the clients specify their host? If they use the [Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) header, then you could get it with SQL like

```sql
create or replace function current_host() returns text as $$
  select nullif(current_setting('request.header.host', true), '')::text;
$$ language sql;
```
(You can get any header with `request.header.<any>`)

**Reply by aarvay** [reply]
Hey @steve-chavez,

current_hostname()` = hostname of wherever the request is coming from. From the MDN docs, what I understand is that the host header refers to where the request is going to

Example:

hosts table: 
| id |                name              |
| -- | ---------------------- |
|  1  | tenant1.example.com  |

 Basically trying to use postgres as a multi-tenant db. A request comes from tenant1.example.com i want to access that in a variable so I can set policies based on that. Each tenant should be able to access/modify only their content.
 
 is this already available via current_setting("request...") or should I set it as a header?

**Reply by steve-chavez** [reply]
You're right, I confused the Host header with the [Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) header. I think that one should do.

> is this already available via current_setting("request...") or should I set it as a header?

Use `select nullif(current_setting('request.header.origin', true), '')::text;`, for the `Origin` header.

**Reply by aarvay** [reply]
`request.header.host` seems to work. thanks @steve-chavez
